orchagent/aclorch.cpp

```diff
@@ -1725,17 +1730,114 @@ void AclOrch::doAclRuleTask(Consumer &consumer)
+        string op = kfvOp(t);
+
+        SWSS_LOG_INFO("doAclTablePortUpdateTask: OP: %s, port_alias: %s", op.c_str(), port_alias.c_str());
+
+        if (op == SET_COMMAND)
+        {
+            for (auto itmap : m_AclTables)
+            {
+                auto table = itmap.second;
+                if (table.pendingPortSet.find(port_alias) != table.pendingPortSet.end())
+                {
+                    SWSS_LOG_INFO("found the port: %s in ACL table: %s pending port list, bind it to ACL table.", port_alias.c_str(), table.description.c_str());
+
+                    bool suc = processPendingPort(table, port_alias, [&](sai_object_id_t portOid) {
+                        table.link(portOid);
+                    });
+
+                    if (!suc)
+                    {
+                        SWSS_LOG_ERROR("Failed to bind the ACL table: %s to port: %s", table.description.c_str(), port_alias.c_str());
+                    }
+                    else
+                    {
+                        table.pendingPortSet.erase(port_alias);
+                        SWSS_LOG_NOTICE("port: %s bound to ACL table table: %s, remove it from pending list", port_alias.c_str(), table.description.c_str());
+                    }
+                }
+            }
+        }
+        else if (op == DEL_COMMAND)
+        {
+            for (auto itmap : m_AclTables)
+            {
+                auto table = itmap.second;
+                if (table.portSet.find(port_alias) != table.portSet.end())
+                {
+                    table.pendingPortSet.emplace(port_alias);
+                    SWSS_LOG_WARN("Add deleted port: %s to the pending list of ACL table: %s", port_alias.c_str(), table.description.c_str());
+                }
+            }
+        }
+        else
+        {
+            SWSS_LOG_ERROR("Unknown operation type %s", op.c_str());
+        }
+        it = consumer.m_toSync.erase(it);
+    }
+}
+
+sai_object_id_t AclOrch::getValidPortId(string alias, Port port)
+{
+    SWSS_LOG_ENTER();
+
+    sai_object_id_t port_id;
+
+    switch (port.m_type)
+    {
+    case Port::PHY:
+        if (port.m_lag_member_id != SAI_NULL_OBJECT_ID)
+        {
+            SWSS_LOG_ERROR("Failed to process port. Bind table to LAG member %s is not allowed", alias.c_str());
+            port_id = SAI_NULL_OBJECT_ID;
+        }
+        else
+        {
+            port_id = port.m_port_id;
+        }
+        break;
+    case Port::LAG:
+        port_id = port.m_lag_id;
+        break;
+    case Port::VLAN:
+        port_id = port.m_vlan_info.vlan_oid;
+        break;
+    default:
+      SWSS_LOG_ERROR("Failed to process port. Incorrect port %s type %d", alias.c_str(), port.m_type);
+      port_id = SAI_NULL_OBJECT_ID;
+      break;
+    }
+
+    return port_id;
+}
+
+bool AclOrch::processPorts(AclTable &aclTable, string portsList, std::function<void (sai_object_id_t)> inserter)
+{
+    SWSS_LOG_ENTER();
+
+    sai_object_id_t port_id;
```

Comment:
since port_id is only used in the for loop, let's move the definition into the for loop.

Reply:
the principle here is to define the variable when you use it.

---
In reply to: [190211288](https://github.com/Azure/sonic-swss/pull/494#discussion_r190211288) [](ancestors = 190211288)